File: client/assets/scripts/protocol/MsgType.ts
/**
 * MsgType.ts - 消息类型 ID 常量
 * 与服务端 pb/msgtype.go 保持一致
 */

// === 系统消息 ===
export const MSG_PING = 9001;
export const MSG_PONG = 9002;

// === 登录 C2S ===
export const MSG_C2S_GUEST_LOGIN = 1001;
export const MSG_C2S_WECHAT_LOGIN = 1002;
export const MSG_C2S_PHONE_LOGIN = 1003;

// === 登录 S2C ===
export const MSG_S2C_LOGIN_RESULT = 2001;

// === 房间/匹配 C2S ===
export const MSG_C2S_QUICK_MATCH = 1101;
export const MSG_C2S_CANCEL_MATCH = 1102;
export const MSG_C2S_CREATE_ROOM = 1103;
export const MSG_C2S_JOIN_ROOM = 1104;
export const MSG_C2S_LEAVE_ROOM = 1105;
export const MSG_C2S_READY = 1106;
export const MSG_C2S_CANCEL_READY = 1107;

// === 房间/匹配 S2C ===
export const MSG_S2C_MATCH_UPDATE = 2101;
export const MSG_S2C_ROOM_JOINED = 2102;
export const MSG_S2C_PLAYER_JOINED = 2103;
export const MSG_S2C_PLAYER_LEFT = 2104;
export const MSG_S2C_READY_UPDATE = 2105;

// === 游戏 C2S ===
export const MSG_C2S_FLIP_CARD = 1201;
export const MSG_C2S_BID = 1202;
export const MSG_C2S_PASS_BID = 1203;
export const MSG_C2S_FORFEIT = 1204;
export const MSG_C2S_SET_BOTTOM = 1205;
export const MSG_C2S_CALL_PARTNER = 1206;
export const MSG_C2S_PLAY_CARD = 1207;
export const MSG_C2S_AUTO_PLAY = 1208;
export const MSG_C2S_CHAT = 1209;

// === 游戏 S2C ===
export const MSG_S2C_ROOM_STATE = 2201;
export const MSG_S2C_DEAL_CARDS = 2202;
export const MSG_S2C_FLIP_RESULT = 2203;
export const MSG_S2C_BID_UPDATE = 2204;
export const MSG_S2C_BID_RESULT = 2205;
export const MSG_S2C_ALL_PASSED = 2206;
export const MSG_S2C_BOTTOM_CARDS = 2207;
export const MSG_S2C_PARTNER_CALLED = 2208;
export const MSG_S2C_TURN = 2209;
export const MSG_S2C_CARD_PLAYED = 2210;
export const MSG_S2C_ROUND_RESULT = 2211;
export const MSG_S2C_GAME_RESULT = 2212;
export const MSG_S2C_GAME_START = 2213;
export const MSG_S2C_ERROR = 2999;

File: server/internal/pkg/pb/msgtype.go
package pb

// 消息类型 ID 常量
// C2S: 1xxx 客户端→服务端
// S2C: 2xxx 服务端→客户端
// 系统: 9xxx 心跳等
const (
	// === 系统消息 ===
	MsgPing uint32 = 9001
	MsgPong uint32 = 9002

	// === 登录 C2S ===
	MsgC2SGuestLogin uint32 = 1001 // 游客登录
	MsgC2SWechatLogin uint32 = 1002
	MsgC2SPhoneLogin  uint32 = 1003

	// === 登录 S2C ===
	MsgS2CLoginResult uint32 = 2001

	// === 房间/匹配 C2S ===
	MsgC2SQuickMatch   uint32 = 1101
	MsgC2SCancelMatch  uint32 = 1102
	MsgC2SCreateRoom   uint32 = 1103
	MsgC2SJoinRoom     uint32 = 1104
	MsgC2SLeaveRoom    uint32 = 1105
	MsgC2SReady        uint32 = 1106
	MsgC2SCancelReady  uint32 = 1107

	// === 房间/匹配 S2C ===
	MsgS2CMatchUpdate  uint32 = 2101
	MsgS2CRoomJoined   uint32 = 2102
	MsgS2CPlayerJoined uint32 = 2103
	MsgS2CPlayerLeft   uint32 = 2104
	MsgS2CReadyUpdate  uint32 = 2105

	// === 游戏 C2S ===
	MsgC2SFlipCard     uint32 = 1201
	MsgC2SBid          uint32 = 1202
	MsgC2SPassBid      uint32 = 1203
	MsgC2SForfeit      uint32 = 1204
	MsgC2SSetBottom    uint32 = 1205
	MsgC2SCallPartner  uint32 = 1206
	MsgC2SPlayCard     uint32 = 1207
	MsgC2SAutoPlay     uint32 = 1208
	MsgC2SChat         uint32 = 1209

	// === 游戏 S2C ===
	MsgS2CRoomState      uint32 = 2201 // 全量同步（断线重连）
	MsgS2CDealCards      uint32 = 2202
	MsgS2CFlipResult     uint32 = 2203
	MsgS2CBidUpdate      uint32 = 2204
	MsgS2CBidResult      uint32 = 2205
	MsgS2CAllPassed      uint32 = 2206
	MsgS2CBottomCards    uint32 = 2207
	MsgS2CPartnerCalled  uint32 = 2208
	MsgS2CTurn           uint32 = 2209
	MsgS2CCardPlayed     uint32 = 2210
	MsgS2CRoundResult    uint32 = 2211
	MsgS2CGameResult     uint32 = 2212
	MsgS2CGameStart      uint32 = 2213 // 游戏开始通知
	MsgS2CError          uint32 = 2999 // 错误消息
)
